src/js/index.js

```diff
@@ -0,0 +1,102 @@
+"use strict";
+// test
+
+const $total = document.querySelector("#total");
+const $digits = document.querySelector(".digits");
+const $operations = document.querySelector(".operations");
+const $modifier = document.querySelector(".modifier");
+
+let isOperationClicked = false;
+let isOperator2In = false; // 두번째 인자가 들어왔는지 체크
+let operator1 = 0;
+let operator2 = 0;
+let operation = "";
+let result = 0;
+
+const initCalculator = () => {
+  isOperationClicked = false;
+  isOperator2In = false;
+  operator1 = 0;
+  operator2 = 0;
+  operation = "";
+  result = 0;
+  $total.textContent = "0";
+};
+
+// 결과창에 보여줄 숫자 제어
+const showTotalNumber = (currentTotal, nextNumber) => {
+  let newTotal = "";
+
+  if (isOperationClicked && !isOperator2In) {
+    newTotal = nextNumber;
+    isOperator2In = true;
+  } else if (isOperationClicked && isOperator2In) {
+    newTotal = currentTotal + nextNumber;
+  } else {
+    if (currentTotal === "0") {
+      newTotal = nextNumber;
+    } else {
+      newTotal = currentTotal + nextNumber;
+    }
+  }
+
+  return newTotal;
+};
+
+const calculate = (currentOperation) => {
+  // 연산자 연속 클릭 방지
+  if (isOperationClicked && currentOperation !== "=") {
+    return;
+  } else if (currentOperation === "=") {
+    operator2 = parseInt($total.textContent);
+
+    switch (operation) {
+      case "+":
+        result = operator1 + operator2;
+        break;
+      case "-":
+        result = operator1 - operator2;
+        break;
+      case "X":
+        result = operator1 * operator2;
+        break;
+      case "/":
+        result = operator1 / operator2;
+        break;
+    }
+    operator1 = result; // 지금까지 결과값을 op1에 저장한다
+    $total.textContent = Math.floor(result);
+
+    isOperator2In = false;
+    isOperationClicked = false;
+  } else {
+    operation = currentOperation;
+    operator1 = parseInt($total.textContent);
+
+    isOperationClicked = true;
+  }
+};
```

Comment:
```suggestion
const calculate = (currentOperation) => {
  if (isOperationClicked && currentOperation !== '=') {
    return
  }
  if (['+', '-', 'X', '/'].includes(currentOperation)) {
    peration = currentOperation
    operator1 = parseInt($total.textContent)
    isOperationClicked = true
  }
  if (currentOperation === '=') {
    operator2 = parseInt($total.textContent)

    switch (operation) {
      case '+':
        result = operator1 + operator2
        break
      case '-':
        result = operator1 - operator2
        break
      case 'X':
        result = operator1 * operator2
        break
      case '/':
        result = operator1 / operator2
        break
    }
    operator1 = result
    $total.textContent = Math.floor(result)
    isOperator2In = false
    isOperationClicked = false
  }
}
```
1. early return을 사용하고나서는 else if 대신 if문으로 분기를 나눠도 좋을것같습니다:)
로직을 확인하니 else의 경우가 ['+', '-', '/', '*'] 연산의 경우에 해당하는 경우임을 알았습니다. 좀 더 의미를 명시적으로 전달하기 위해서 include를 활용해보았습니다. 

2. 예시를 위해 include를 사용하였지만 예시 코드에 단점이 있습니다. 만약 배열에 새로운 피연산자가 추가된다고 가정하면 조건문 배열도 바꿔야하고 case문도 추가해야합니다.  이는 실수를 유발할수있는 형태이기에, 상수로 빼서 활용하면 조금 더 안전하게 사용할 수 있을 것 같아요:)